rats-tls/src/include/enclave-tls/api.h

```diff
@@ -64,6 +64,42 @@ typedef struct {
 	} quote_sgx_ecdsa;
 } enclave_tls_conf_t;
 
+typedef struct etls_sgx_evidence {
+	uint8_t *mrEnclave;
+	uint8_t *mrSigner;
+	uint32_t productId;
+	uint32_t securityVersion;
+	uint8_t *attributes;
+	size_t collateral_size;
+	char *collateral;
+} etls_sgx_evidence_t;
+
+typedef struct etls_tdx_evidence {
+	/* TODO */
+} etls_tdx_evidence_t;
+
+/* The public_key, user_data_size and user_data are needed to include in hash. */
+typedef struct ehd {
+	void *public_key;
+	int user_data_size;
+	char *user_data;
+	int unhashed_size;
+	char *unhashed;
+} ehd_t;
+
+typedef enum { SGX_ECDSA = 1, TDX } enclave_evidence_type_t;
+
+typedef struct etls_evidence {
+	enclave_evidence_type_t type;
+	ehd_t ehd;
+	int quote_size;
+	char *quote;
+	union {
+		etls_sgx_evidence_t sgx;
```

Comment:
rename to `etls_sgx_ecdsa_evidence_t sgx_ecdsa;`

Reply:
etls_sgx_evidence_t is not just for ecdsa, it can be used later by Local attestation as scalable.